### PR TITLE
Adding support for importing/exporting polylines in COCO/CVAT formats

### PR DESCRIPTION
**Pull request by brimoor**
The COCO detection format supports storing instance segmentation masks.

And, the CVAT image/video formats support storing polylines, polygons, and keypoints.

This PR adds support for importing/exporting this data when working with the `COCODetectionDataset`, `CVATImageDataset`, and `CVATVideoDataset` formats, respectively, in FiftyOne.

Example of reading/writing segmentation masks in COCO format:

```py
import fiftyone as fo

IMAGE_PATH = "/path/to/image.png"
EXPORT_DIR = "/tmp/coco-export"
MASK_SIZE = (64, 64)

sample = fo.Sample(filepath=IMAGE_PATH)

polylines = [
    fo.Polyline(label="giraffe", points=[(0.1, 0.1), (0.4, 0.4), (0.1, 0.4)], closed=True, filled=True),
    fo.Polyline(label="rhino", points=[(0.6, 0.6), (0.9, 0.9)], closed=False, filled=False),
]

detections = [p.to_detection(MASK_SIZE) for p in polylines]

sample["gt_polys"] = fo.Polylines(polylines=polylines)
sample["gt_objs"] = fo.Detections(detections=detections)

dataset = fo.Dataset()
dataset.add_sample(sample)

dataset.export(
    export_dir=EXPORT_DIR,
    dataset_type=fo.types.COCODetectionDataset,
    overwrite=True,
)

dataset2 = fo.Dataset.from_dir(EXPORT_DIR, fo.types.COCODetectionDataset)

# Visualize `dataset`
session = fo.launch_app(dataset)

# Now visualize `dataset2`
session.dataset = dataset2
```

Example of reading/writing polylines/polygons/keypoints in CVAT image/video formats:

```py
import fiftyone as fo

frame = fo.Frame(
    gt_attrs=fo.Classification(label="sunny"),
    gt_objs=fo.Detections(
        detections=[
            fo.Detection(label="cat", bounding_box=[0.4, 0.4, 0.2, 0.2]),
            fo.Detection(label="dog", bounding_box=[0.7, 0.7, 0.2, 0.2]),
        ]
    ),
    gt_polys=fo.Polylines(
        polylines=[
            fo.Polyline(label="giraffe", points=[(0.1, 0.1), (0.2, 0.2), (0.1, 0.2)], closed=True, filled=True),
            fo.Polyline(label="rhino", points=[(0.3, 0.3), (0.3, 0.5), (0.5, 0.5)], closed=False, filled=False),
        ]
    ),
    gt_kpts=fo.Keypoints(
        keypoints=[
            fo.Keypoint(label="human", points=[(0.7, 0.7), (0.8, 0.8), (0.9, 0.9)]),
        ]
    ),
)

#
# CVATVideoDataset
#

VIDEO_PATH = "/path/to/video.mp4"
EXPORT_DIR = "/tmp/cvat-video-export"

sample = fo.Sample(filepath=VIDEO_PATH)

sample.frames[1] = frame

dataset = fo.Dataset()
dataset.add_sample(sample)

dataset.export(
    export_dir=EXPORT_DIR,
    dataset_type=fo.types.CVATVideoDataset,
    overwrite=True,
)

dataset2 = fo.Dataset.from_dir(
    EXPORT_DIR, fo.types.CVATVideoDataset, label_field="gt"
)

print(dataset)
print(dataset2)

print(dataset.first().frames)
print(dataset2.first().frames)

#
# CVATImageDataset
#

IMAGE_PATH = "/path/to/image.png"
EXPORT_DIR = "/tmp/cvat-image-export"

sample = fo.Sample.from_frame(frame, IMAGE_PATH)

dataset = fo.Dataset()
dataset.add_sample(sample)

dataset.export(
    export_dir=EXPORT_DIR,
    dataset_type=fo.types.CVATImageDataset,
    overwrite=True,
)

dataset2 = fo.Dataset.from_dir(
    EXPORT_DIR, fo.types.CVATImageDataset, label_field="gt"
)

print(dataset)
print(dataset2)

print(dataset.first())
print(dataset2.first())
```

### CODE REVIEW

**Review by benjaminpkane — APPROVED** [review]
LGTM! I can't say I read every line, but I'd like to merge into my optimizations and test more from there.

I will note that I got an error in from the first snippet before I reduced the mask size to `(32, 32)`. I suppose the image I used was too small?
```
Traceback (most recent call last):
  File "ttt.py", line 28, in <module>
    dataset2 = fo.Dataset.from_dir(EXPORT_DIR, fo.types.COCODetectionDataset)
  File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 1441, in from_dir
    **kwargs,
  File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 984, in add_dir
    add_info=add_info,
  File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 1023, in add_importer
    add_info=add_info,
  File "/home/ben/code/fiftyone/fiftyone/utils/data/importers.py", line 208, in import_samples
    samples, expand_schema=expand_schema, num_samples=num_samples
  File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 663, in add_samples
    for batch in fou.iter_batches(samples, batch_size):
  File "/home/ben/code/fiftyone/fiftyone/core/utils.py", line 536, in iter_batches
    chunk = tuple(itertools.islice(it, batch_size))
  File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 224, in __next__
    self.load_segmentations,
  File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 938, in _coco_objects_to_detections
    load_segmentation=load_segmentations,
  File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 537, in to_detection
    self.segmentation, self.bbox, frame_size
  File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 992, in _coco_segmentation_to_mask
    mask_utils.frPyObjects(segmentation, height, width)
  File "pycocotools/_mask.pyx", line 292, in pycocotools._mask.frPyObjects
IndexError: list index out of range
```

**Comment by brimoor** [comment]
> LGTM! I can't say I read every line, but I'd like to merge into my optimizations and test more from there.
> 
> I will note that I got an error in from the first snippet before I reduced the mask size to `(32, 32)`. I suppose the image I used was too small?
> 
> ```
> Traceback (most recent call last):
>   File "ttt.py", line 28, in <module>
>     dataset2 = fo.Dataset.from_dir(EXPORT_DIR, fo.types.COCODetectionDataset)
>   File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 1441, in from_dir
>     **kwargs,
>   File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 984, in add_dir
>     add_info=add_info,
>   File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 1023, in add_importer
>     add_info=add_info,
>   File "/home/ben/code/fiftyone/fiftyone/utils/data/importers.py", line 208, in import_samples
>     samples, expand_schema=expand_schema, num_samples=num_samples
>   File "/home/ben/code/fiftyone/fiftyone/core/dataset.py", line 663, in add_samples
>     for batch in fou.iter_batches(samples, batch_size):
>   File "/home/ben/code/fiftyone/fiftyone/core/utils.py", line 536, in iter_batches
>     chunk = tuple(itertools.islice(it, batch_size))
>   File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 224, in __next__
>     self.load_segmentations,
>   File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 938, in _coco_objects_to_detections
>     load_segmentation=load_segmentations,
>   File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 537, in to_detection
>     self.segmentation, self.bbox, frame_size
>   File "/home/ben/code/fiftyone/fiftyone/utils/coco.py", line 992, in _coco_segmentation_to_mask
>     mask_utils.frPyObjects(segmentation, height, width)
>   File "pycocotools/_mask.pyx", line 292, in pycocotools._mask.frPyObjects
> IndexError: list index out of range
> ```

Hmm let me think if some error checking needs to go in there. Thanks for pointing out that issue

File: fiftyone/utils/coco.py [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 """
 from collections import defaultdict
 from datetime import datetime
+from itertools import groupby
 import logging
 import os
+import warnings
+
+import numpy as np
+from skimage import measure
```
Comment (lethosor) [inline fiftyone/utils/coco.py:17]: `skimage` appears to be a new dependency - is it optional? It's causing zoo documentation generation to fail now, so it should at least be documented somewhere, even if it's not required.
https://github.com/voxel51/fiftyone/runs/1250898309#step:9:176

Comment (benjaminpkane) [reply fiftyone/utils/coco.py:17]: Yes, I forgot to mention that in my review. My mistake.

**Pull request merged**